march_hardware/include/march_hardware/imotioncube/motor_controller.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Project March.
+
+#include "actuation_mode.h"
+#include "motor_controller_state.h"
+#include <string>
+
+namespace march
+{
+class MotorController
+{
+public:
+  virtual double getAngleRadAbsolute() = 0;
+  virtual double getAngleRadIncremental() = 0;
+  virtual double getVelocityRadAbsolute() = 0;
+  virtual double getVelocityRadIncremental() = 0;
+  virtual int16_t getTorque() = 0;
+  virtual MotorControllerStates getStates() = 0;
+
+  virtual ActuationMode getActuationMode() const = 0;
+  virtual uint16_t getSlaveIndex() const = 0;
+
+  virtual float getMotorCurrent() = 0;
+  virtual float getMotorControllerVoltage() = 0;
+  virtual float getMotorVoltage() = 0;
+  virtual bool getIncrementalMorePrecise() const = 0;
```

Comment:
This also seems like a very specific function that should not be inside the motor controller class. When I read this method call I cannot tell what it would do generally on most controllers.

Reply:
Added a docstring with explanation.